data-service/src/main/java/uk/gov/gchq/palisade/service/data/model/AuditableAuthorisedDataRequest.java

```diff
@@ -0,0 +1,157 @@
+
+        this.dataRequest = dataRequest;
+        this.authorisedDataRequest = authorisedDataRequest;
+        this.auditErrorMessage = auditErrorMessage;
+    }
+
+    @Generated
+    public DataRequest getDataRequest() {
+        return dataRequest;
+    }
+
+    @Generated
+    public AuthorisedDataRequest getAuthorisedDataRequest() {
+        return authorisedDataRequest;
+    }
+
+
+    @Generated
+    public AuditErrorMessage getAuditErrorMessage() {
+        return auditErrorMessage;
+    }
+
+    /**
+     * The static builder
+     */
+    public static class Builder {
+
+        /**
+         * The starter method for the Builder class.
+         *
+         * @return the composed immutable object
+         */
+        public static IDataRequest create() {
+            return dataRequest -> (authorisedData, error) -> new AuditableAuthorisedDataRequest(dataRequest, authorisedData, error);
+        }
+
+        /**
+         * Adds the {@link DataRequest} to the container
+         */
+        public interface IDataRequest {
+            /**
+             * Adds the dataRequest.
+             *
+             * @param dataRequest initial request data
+             * @return interface {@link IDataResponse} for the next step of the build
+             */
+            IDataResponse withDataRequest(DataRequest dataRequest);
+
+        }
+
+        /**
+         * Compose wityh either {@link AuthorisedDataRequest} or {@link AuditErrorMessage} to create the
+         * AuditableAuthorisedDataRequest
+         */
+        public interface IDataResponse {
+            /**
+             * Adds the data from the
+             *
+             * @param authorisedDataRequest for the resources
+             * @return class {@link AuditableAuthorisedDataRequest} for the final step in the build.
+             */
+            default AuditableAuthorisedDataRequest withAuthorisedData(AuthorisedDataRequest authorisedDataRequest) {
+                return withAuthorisedDataAndError(authorisedDataRequest, null);
+            }
+
+            /**
+             * Adds the data from the
+             *
+             * @param auditErrorMessage for the error that occurred
+             * @return class {@link AuditableAuthorisedDataRequest} for the final step in the build.
+             */
+            default AuditableAuthorisedDataRequest withAuditErrorMessage(AuditErrorMessage auditErrorMessage) {
+                return withAuthorisedDataAndError(null, auditErrorMessage);
+            }
+
+            /**
+             * Adds both the {@link AuthorisedDataRequest } and the {@link AuditErrorMessage} for the creation of the object
+             * @param authorisedDataRequest  value or null
+             * @param auditErrorMessage value or null
+             * @return value object
+             */
+            AuditableAuthorisedDataRequest withAuthorisedDataAndError(AuthorisedDataRequest authorisedDataRequest, AuditErrorMessage auditErrorMessage);
```

Comment:
is there ever a need to have both the request and the error? surley its a or, rather than an and?

Reply:
This is something that has been done in the other services where the `withResponse` calls a `withResponseAndError` method but puts a `null` for the AuditMessage. I would leave this in for consistency between services.